adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -182,6 +183,13 @@ trait AkkaHttpAdapter {
               case ObjectValue((fieldName, StreamValue(stream)) :: Nil) =>
                 stream
                   .foreach(item => sendMessage(sendTo, messageId, ObjectValue(List(fieldName -> item)), result.errors))
+                  .onExit {
+                    case Success(_) =>
+                      IO.fromFuture(_ => sendTo.offer(TextMessage(s"""{"type":"complete","id":"$messageId"}"""))).orDie
+                    case Failure(_) =>
+                      IO.fromFuture(_ => sendTo.offer(TextMessage(json.encodeWSError(messageId, "Upstream failed"))))
```

Comment:
In the case of a clean `stop` where we cancel the stream execution, I think there will be a spurious error message, no?